### ISSUE TITLE
[ASTImporter] Class with repeated friends are rejected when re-imported into initial empty Context

### ISSUE THREAD

**Issue by danix800** [llvm/llvm-project#64425]
```c
struct foo { friend class X; friend class X; }; // FromContext
```
only one friend is imported, similar to the following:
```c
struct foo { friend class X; }; // ToContext
```
but when imported again, `struct foo` in FromContext is reported as
not equivalent to `struct foo` in ToContext, thus rejected.

**Comment by danix800** [comment]
Fix proposed: https://reviews.llvm.org/D157114

**Comment by danix800** [comment]
Misunderstanding of how repeated friends are imported.
